packages/data-connect/src/api/DataConnect.ts

```diff
@@ -71,11 +70,15 @@ const FIREBASE_DATA_CONNECT_EMULATOR_HOST_VAR =
  * @internal
  */
 export function parseOptions(fullHost: string): TransportOptions {
-  const [protocol, hostName] = fullHost.split('://');
-  const isSecure = protocol === 'https';
-  const [host, portAsString] = hostName.split(':');
-  const port = Number(portAsString);
-  return { host, port, sslEnabled: isSecure };
+  if (fullHost.startsWith('http://') || fullHost.startsWith('https://')) {
```

Comment:
Should we trim() the fullHost string before comparing startsWith.

Reply:
Done